File: javascript/reverse_string.js
// Use Array Methods
// function reverseString1(str) {
//   return str.split("").reverse().join("");
// }

// Using While Loop

function reverseString(str) {
  let reverse_str = "", string = str;
  while (string.length > 0) {
    reverse_str = string[0] + reverse_str;
    string = string.slice(1)
  }
  return reverse_str
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: ruby/reverse_string.rb
def reverse_string(str)
  reverse_str = ""

  while str.length > 0
    reverse_str = str[0]+reverse_str
    str = str[1...]
  end

  reverse_str
end


if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution